### PR TITLE
RHCLOUD-29917 | fix: service account getting results from custom default access groups

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
## Link(s) to Jira
- [[RHCLOUD-29917]](https://issues.redhat.com/browse/RHCLOUD-29917)

## Description of Intent of Change(s)
The custom default access groups were making the access endpoint to return certain permissions for service accounts, when they should not be getting any.
### How can the bug be exploited and fix confirmed?
1. Create a new service account.
2. Call the `/access/?application=remediations` endpoint with the account's credentials and confirm that the data key is empty.
3. Add a new role to the default access group.
4. Call the access endpoint again, and see how it returns permissions when it really shouldn't.

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by vbelchio** [comment]
QE: Passed

File: rbac/management/utils.py [hunk 1]
```diff
@@ -143,17 +143,19 @@ def groups_for_principal(principal: Principal, tenant, **kwargs):
...
-    admin_default_group_set = Group.admin_default_set().filter(tenant=tenant)
-
-    # Same as above, the default admin group should only be fetched for the users.
-    if not admin_default_group_set and principal.type == "user":
-        admin_default_group_set = Group.admin_default_set().filter(tenant=public_tenant)
+    # Service accounts should not get permissions from any of the default groups. The customers need to explicitly add
+    # the service accounts to new RBAC groups and assign them permissions.
+    if principal.type == "service-account":
+        admin_default_group_set = Group.objects.filter(admin_default=False).filter(tenant=tenant)
+        platform_default_group_set = Group.objects.filter(platform_default=False).filter(tenant=tenant)
```
Comment (astrozzc) [inline rbac/management/utils.py:151]: According to the description, admin_default_group_set and platform_default_group_set should be None for the service account?
Group.objects.filter(admin_default/platform_default=False) would return all custom groups even the account is not assigned for. Therefore, the assigned_group_set is enough for the service account?

**Review by astrozzc — APPROVED** [review]
LGTM!

**Review by abaiken — APPROVED** [review]

**Pull request merged**